text/0012-styleguide-javascript-components.md

```diff
@@ -0,0 +1,64 @@
+# Styleguide javascript components
+- Start Date: 2019/07/24
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/12
+- Previous, defunct RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/9
+
+## Decision, including impact on distributions
+The [OpenMRS styleguide](/text/0008-styleguide.md) provides a CSS-first API to be used by all
+frontend code at OpenMRS. Framework-specific Javascript components (for React, Vue, and/or Angular) will be
+implemented when convenient, but are not required for using the styleguide's global css classes.
```

Comment:
I guess I'm not totally sure what this means. "Will be implemented when convenient" means that Styleguide managers will implement these components for certain specific elements? Or means that people are welcome to implement components, but that doing so will not be necessary to use the OpenMRS Styleguide? (That "[javascript components] are not required for using the styleguide's global css classes" seems like it should be a truism, but maybe there are libraries that have global CSS classes that are only usable with javascript components?)

Reply:
Yeah good question -- what I'm trying to emphasize is that the styleguide's primary API is the css, not the js components. The reason for this is that making cross-framework js components for a styleguide is quite difficult.

Most of the styleguide won't have any js components at all. But for some things within the styleguide, such as multiselects, we will need js components. What this bullet point in the RFC is trying to say is that we can't commit to having a js component in every possible framework for multiselects. We will build the js components only as needed in each of the frameworks.

Reply:
How about

"For the few elements for which JavaScript components are required (e.g. multiselect), the Styleguide will provide a specification (and a reference implementation?). Implementations of that specification for other frameworks will be integrated into the Styleguide whenever they happen to be written."

Reply:
👍 updated this based on your feedback.